### ISSUE TITLE
MOF compiler support for finding REF/Embedded classes and not creating inconsistent repo without them

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbem#1160]
The mof_compiler includes support for finding dependent classes that is aimed at finding things like:

1. Classes that are dependent on the EmbeddedInstance qualifier but that the class has not been compiled already

2. Classes that where the defined superclass has not been compiled.

3. Reference properties where the ReferenceClass has not been defined.

This is part of the method p_mp_createClass which upon receiving an exception to a createclass tries to
fix issues with a)superclass not yet compiled, b) class for embeddedInstance not yet compiled, c) reference classes not yet compiled

However, while the code is in place, it depends on the mof_compiler actually creating an exception when the preconditions have not been met.  This works for superclasses but does not appear to work for either the embeddedInstance qualifier or for reference properties. 

We need to add exceptions for the following:

For the reference property need to add test for existence of the compiled reference class.
The following is an example:

def p_referenceDeclaration

```
        try:
            qual = self.classes[self.default_namespace][cname]
        except KeyError:
            if self.conn is None:
                raise CIMError(CIM_ERR_NOT_FOUND, cname)

```
For the EmbeddedInstanceQualifier:

Add a similar test to p_propertyDeclaration functions to test for EmbeddedInstance Qualifier  and then test for existence of the corresponding class

These exceptions would then be caught 

At least that is my best guess right now.

**Comment by andy-maier** [comment]
**Discussion point:** Target release?

**Comment by andy-maier** [comment]
Bottom line, the issue is twofold:
1. the MOF compiler succeeds even though it creates an inconsistent repository (where REF classes and EmbeddedInstance-qualified classes are used but not present). This part is a bug.
2. the MOF compiler does not automatically resolve the dependent classes for REF and EmbeddedInstance. This part is a missing feature.

**Comment by andy-maier** [comment]
We agreed that we can release 0.12 without fixing this.

**Comment by andy-maier** [comment]
Fixed by PR #1217.
